src/cluster_import.c

```diff
@@ -0,0 +1,1948 @@
+            }
+            target_node = clusterLookupNode(c->argv[4]->ptr, CLUSTER_NAMELEN);
+            if (!target_node) {
+                serverLog(
+                    LL_WARNING,
+                    "Received CLUSTER SYNCSLOTS SNAPSHOT from client %llu with an unknown node ID. "
+                    "Failing link.",
+                    (unsigned long long)c->id);
+                error = 1;
+                break;
+            }
+        }
+        if (!strcasecmp(c->argv[i]->ptr, "linkname")) {
+            if (link_name || i + 1 >= c->argc || sdslen(c->argv[i + 1]->ptr) != CLUSTER_NAMELEN) {
+                serverLog(
+                    LL_WARNING,
+                    "Malformatted or missing link name in CLUSTER SYNCSLOTS SNAPSHOT from client "
+                    "%llu. Failing link.",
+                    (unsigned long long)c->id);
+                error = 1;
+                break;
+            }
+            link_name = c->argv[i + 1]->ptr;
+        }
+        if (!strcasecmp(c->argv[i]->ptr, "slotsrange")) {
+            slot_ranges = parseSlotRanges(c, i + 1, &source_node, &err);
+            if (err) {
+                serverLog(
+                    LL_WARNING,
+                    "Failed to parse slot range provided by CLUSTER SYNCSLOTS SNAPSHOT from client "
+                    "%llu: %s. Failing link.",
+                    (unsigned long long)c->id, err);
+                sdsfree(err);
+                error = 1;
+                break;
+            }
+            if (source_node != server.cluster->myself) {
+                serverLog(LL_WARNING,
+                          "Received CLUSTER SYNCSLOTS SNAPSHOT from client %llu, "
+                          "but I am not the owner of the requested slots. Failing link.",
+                          (unsigned long long)c->id);
+                error = 1;
+                break;
+            }
+            /* The last argument should be SLOTSRANGE */
+            break;
+        }
+    }
+    if (!error && (!target_node || !link_name || !slot_ranges)) {
+        serverLog(LL_WARNING,
+                  "Missing >= 1 required argument in CLUSTER SYNCSLOTS SNAPSHOT from client %llu. "
+                  "Failing link.",
+                  (unsigned long long)c->id);
+        error = 1;
+    }
+    if (error) {
+        if (slot_ranges) {
+            listRelease(slot_ranges);
+        }
+        sendFailAndCloseAfterReply(c);
+        return;
+    }
+
+    slotMigrationLink *link = createSlotExportLink(c, target_node->name, link_name, slot_ranges);
+    listAddNodeHead(server.cluster->slot_migration_links, link);
+
+    /* No reply, we will proceed with AOF formatted dump once BGSAVE is able to
+     * succeed */
+    clusterDoBeforeSleep(CLUSTER_TODO_HANDLE_SLOT_MIGRATION);
+}
+
+/* There are three potential triggers for streaming (whichever happens first):
+ *   1. SYNCSLOTS STREAM command
+ *   2. SYNCSLOTS PAUSE command
+ *   3. BGSAVE child process dies
```

Comment:
I don't know, but I associate "a process dies" with something that did not work as intended

```suggestion
 *   3. BGSAVE child process finishes successfully
```

Reply:
Makes sense